src/Okta.DNX.OAuth.ResourceServer/Startup.cs

```diff
@@ -0,0 +1,123 @@
+            }
+
+            builder.AddEnvironmentVariables();
+
+            Configuration = builder.Build();
+
+
+            clientId = Configuration["okta:clientId"] as string;
+            issuer = Configuration["okta:organizationUrl"];
+            authorizationServerIssuer = Configuration["okta:authorizationServerIssuer"];
+            audience = Configuration["okta:audience"] as string;
+        }
+
+        public IConfigurationRoot Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Add framework services.
+            //services.AddApplicationInsightsTelemetry(Configuration);
+
+            //     var scopePolicy = new AuthorizationPolicyBuilder()
+            //.RequireAuthenticatedUser()
+            //.RequireClaim("scp", "call-api")
+            //.Build();
+
+            services.AddAuthentication();
+            services.AddAuthorization(options =>
+            {
+                options.AddPolicy("todo.read",
+                    policy =>
+                    {
+                        policy
+                        .RequireClaim("cid", clientId)
+                       .RequireClaim("http://schemas.microsoft.com/identity/claims/scope", "todolist.read");
+                    }
+                );
+            }
+            );
+
+            services.AddMvc();
+
+            services.AddSingleton<ITodoRepository, TodoRepository>();
+        }
+
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+        {
+            loggerFactory.AddConsole(LogLevel.Trace);
+            loggerFactory.AddDebug();
+
+            //app.UseApplicationInsightsRequestTelemetry();
+
+            //app.UseApplicationInsightsExceptionTelemetry();
+
+
+            TokenValidationParameters tvps = new TokenValidationParameters
+            {
+                ValidateAudience = true,
+                ValidAudience = audience,
+
+                ValidateIssuer = true,
+                ValidIssuer = authorizationServerIssuer,
+
+                ValidateLifetime = true,
+
+
+                ClockSkew = TimeSpan.FromMinutes(5)
+            };
+
+            // Configure the app to use Jwt Bearer Authentication
+            app.UseJwtBearerAuthentication(new JwtBearerOptions
+            {
+                //MetadataAddress is critical for the JwtBearerAuthentication middleware to retrieve the OIDC metadata and be able to perform signing key validation
+                MetadataAddress = authorizationServerIssuer + "/.well-known/openid-configuration",
+                TokenValidationParameters = tvps
+            });
+
+            app.UseMvc();
+        }
+    }
+}
```

Comment:
there is a trick to make the minor style right once, if you are using visual studio, simply return the last `}` then add `}` back, all the indent will be adjusted nicely.